src/stories/TextInput.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useState, useEffect } from "react";
+import InfoIcon from "@mui/icons-material/Info";
+import "./textinput.css";
+import TitleIcon from "@mui/icons-material/Title";
+import KeyboardReturnIcon from "@mui/icons-material/KeyboardReturn";
+
+interface TextInputProps {
+  placeholder: string;
+  isDisabled: boolean;
+  isPassword: boolean;
+  validationRegex: RegExp;
```

Comment:
Accepting only a RegExp object means that this field breaks in Storybook. Consider also accepting a string (and making sure Storybook is treating this field as a string.)